### PR TITLE
Simplify nodejs client and wallet

### LINKED ISSUE #699: Investigate: Nodejs bindings wrapper type
Investigate if we can change the bindings wrappers into something more usable.
Currently its a type wrapped in a struct, that aint pretty

See #679

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

This PR cleans up the client and wallet nodejs rust bindings.

## Links to any relevant issues

Closes #699

### CODE REVIEW

**Review by thibault-martinez — COMMENTED** [review]
Do we want to do something similar with `SecretManagerMethodHandler`? It has internal Arc/RwLock. Or why is this one different?

**Comment by DaughterOfMars** [comment]
They all have additional internal locking. The difference is that the secret manager doesn't have the outer Arc with the `Option` inside to control whether it's created or not

**Comment by thibault-martinez** [comment]
Did you say you needed help to try it or is this completely ready to be reviewed/merged?

**Comment by DaughterOfMars** [comment]
Yeah I need help ensuring the nodejs side is correct.

**Comment by Thoralf-M** [comment]
Maybe enable the CI for pull requests to the 1.1 branch

**Comment by thibault-martinez** [comment]
> Maybe enable the CI for pull requests to the 1.1 branch

They are already enabled

**Comment by Thoralf-M** [comment]
> They are already enabled

Why did nothing run on this PR then?

**Comment by DaughterOfMars** [comment]
Because I did a base change and github is bad

**Comment by DaughterOfMars** [comment]
```Jest has detected the following 1 open handle potentially keeping Jest from exiting:```

I think this is because we are no longer returning a promise, so evidence that nodejs needs changes.

**Comment by kwek20** [comment]
> `Jest has detected the following 1 open handle potentially keeping Jest from exiting:`
> 
> I think this is because we are no longer returning a promise, so evidence that nodejs needs changes.

No i've tried to solve this before using the .finish() on client, but i dont think its actually related to our code

**Comment by Thoralf-M** [comment]
Tests need to be updated because the error message changed

File: bindings/wasm/src/wallet.rs [hunk 1]
```diff
@@ -38,38 +57,34 @@ pub fn create_wallet(options: String) -> Result<WalletMethodHandler, JsValue> {
         .map_err(|e| e.to_string())?;
 
     Ok(WalletMethodHandler {
-        wallet: Arc::new(Mutex::new(Some(wallet_method_handler))),
+        wallet: Arc::new(RwLock::new(Some(wallet_method_handler))),
```
Comment (DaughterOfMars) [inline bindings/wasm/src/wallet.rs]: Why use the std version?

Comment (kwek20) [reply bindings/wasm/src/wallet.rs]: we also do in nodejs. but the std one is not async which is the main reason

Comment (DaughterOfMars) [reply bindings/wasm/src/wallet.rs]: Yeah but we are in async fns...

Comment (kwek20) [reply bindings/wasm/src/wallet.rs]: .client() isnt, just like in rust

File: bindings/nodejs/src/client.rs [hunk 2]
```diff
@@ -60,71 +58,69 @@ impl ClientMethodHandler {
...
+                        let cb = callback.into_inner(&mut cx);
+                        let this = cx.undefined();
+
+                        let args = [
+                            if is_error {
+                                cx.string(response.clone()).upcast::<JsValue>()
+                            } else {
+                                cx.undefined().upcast::<JsValue>()
+                            },
+                            cx.string(response).upcast::<JsValue>(),
```
Comment (Alex6323) [inline bindings/nodejs/src/client.rs]: just for my understanding ... as it's wasn't changed in this PR ... but why do we need line 101?

File: bindings/wasm/src/wallet.rs (hunk 1 above)
Comment (DaughterOfMars) [reply bindings/wasm/src/wallet.rs]:
```rust
pub async fn get_client(method_handler: &WalletMethodHandler) -> Result<ClientMethodHandler, JsValue> {
```

File: bindings/nodejs/src/client.rs (hunk 2 above)
Comment (DaughterOfMars) [reply bindings/nodejs/src/client.rs]:
```
callClientMethod(method, handler, (error: Error, result: string)
```

The TS expects that tuple so this is building that.

**Pull request closed without merging**